src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -6951,9 +6951,19 @@ void CodeGen::genSSE41RoundOp(GenTreeOp* treeNode)
             switch (memBase->OperGet())
             {
                 case GT_LCL_VAR_ADDR:
+                case GT_LCL_FLD_ADDR:
                 {
+                    assert(memBase->isContained());
                     varNum = memBase->AsLclVarCommon()->GetLclNum();
-                    offset = 0;
+                    if (memBase->OperIs(GT_LCL_FLD_ADDR))
```

Comment:
A comment here would be good - i.e. that this path is unexpected and untested, but should be correct in case it is encountered in a non-checked JIT.

Reply:
Thanks, I am planning to write a test that hits that path, if I won't be able to construct it I will add a comment.

Reply:
Added the repro cases and deleted the asserts.